plugins/gateway/kong/plugin.go

```diff
@@ -0,0 +1,179 @@
+	telemetry, err := createTelemetry(kong)
+	if err != nil {
+		_ = kong.Log.Err(fmt.Sprintf("Failed to create telemetry. %v", err))
+		return
+	}
+
+	params := operations.NewPostTelemetryParams().WithBody(telemetry)
+
+	_, err = conf.apiClient.Operations.PostTelemetry(params)
+	if err != nil {
+		_ = kong.Log.Err(fmt.Sprintf("Failed to post telemetry : %v", err))
+	}
+}
+
+func createTelemetry(kong *pdk.PDK) (*models.Telemetry, error) {
+	routedService, err := kong.Router.GetService()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get routed serivce. %v", err)
+	}
+	clientIP, err := kong.Client.GetIp()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get client ip. %v", err)
+	}
+	//clientPort, err := kong.Client.GetPort()
+	//if err != nil {
+	//	return nil, fmt.Errorf("failed to get client port. %v", err)
+	//}
+	destPort := routedService.Port
+	host := routedService.Host
+
+	// Will get the actual path that the request was sent to, not the routed one
+	path, err := kong.Request.GetPath()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request path. %v", err)
+	}
+	reqBody, err := kong.Request.GetRawBody()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request body. %v", err)
+	}
+	resBody, err := kong.ServiceResponse.GetRawBody()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get response body. %v", err)
+	}
+	method, err := kong.Request.GetMethod()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request method. %v", err)
+	}
+
+	statusCode, err := kong.ServiceResponse.GetStatus()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get response status code. %v", err)
+	}
+	scheme, err := kong.Request.GetScheme()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get reuqest scheme. %v", err)
+	}
+	version, err := kong.Request.GetHttpVersion()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request http version. %v", err)
+	}
+	reqHeaders, err := kong.Request.GetHeaders(-1) // default limit of 100 headers
+	if err != nil {
+		return nil, fmt.Errorf("failed to get request headers. %v", err)
+	}
+	// Unlike kong.Response.GetHeaders(), this function will only return headers
+	// that were present in the response from the Service (ignoring headers added
+	// by Kong itself)
+	resHeaders, err := kong.ServiceResponse.GetHeaders(-1) // default limit of 100 headers
+	if err != nil {
+		return nil, fmt.Errorf("failed to get response headers. %v", err)
+	}
+
+	telemetry := models.Telemetry{
+		DestinationAddress:   ":" + strconv.Itoa(destPort), // No destination ip for now
+		DestinationNamespace: "",
+		Request: &models.Request{
+			Common: &models.Common{
+				TruncatedBody: false,
+				Body:          strfmt.Base64(reqBody),
+				Headers:       createHeaders(reqHeaders),
+				Version:       fmt.Sprintf("%f", version),
+			},
+			Host:   parseHost(host),
+			Method: method,
+			Path:   path,
+		},
+		RequestID: "", // TODO from where? do we need?
```

Comment:
not sure that we are using it - you can just generate a uuid if nothing on the request. Kong is not adding some king of req id header?